xtrack/beam_elements/apertures.py

```diff
@@ -232,12 +232,13 @@ def y_closed(self):
         return np.concatenate([yy, np.array([yy[0]])])
 
     def impact_point_and_normal(self, x_in, y_in, z_in,
-                                x_out, y_out, z_out):
+                                x_out, y_out, z_out, particles_class):
```

Comment:
This shouldn't be here...

Reply:
The workaround for this problem ended up being a bigger change. All the relevant changes are in this commit: https://github.com/xsuite/xtrack/pull/312/commits/39e8654997456f3e6b6f23c472f9dc2e224055f5 

The issue was that while [`LimitPolygon_impact_point_and_normal`](https://github.com/xsuite/xtrack/blob/a65498cca524f0c8092fb723a1eaf63d45ca9735/xtrack/beam_elements/apertures_src/limitpolygon.h#L52) does not require accessing `x` and `y` of the particle, [`LimitPolygon_track_local_particle`](https://github.com/xsuite/xtrack/blob/a65498cca524f0c8092fb723a1eaf63d45ca9735/xtrack/beam_elements/apertures_src/limitpolygon.h#L11) does. Therefore that part of the code needs to be disabled for `ParticlesBase`. There already is a flag ` NO_LIMITPOLYGON_TRACK_LOCAL_PARTICLE`, so it can be "enabled" to skip that part. This also required the removal of the default per particle track kernel, which seems sensible as after the introduction of the minitracker we no longer need it anyway.